sdk/spring/azure-spring-boot-test-aad-resource-server-by-filter/src/test/java/com/azure/test/aad/filter/stateful/AADAuthenticationFilterIT.java

```diff
@@ -0,0 +1,152 @@
+    public void init() {
+        singleTenantITHelper = getAADWebApiITHelper(AAD_SINGLE_TENANT_CLIENT_ID, AAD_SINGLE_TENANT_CLIENT_SECRET);
+        multiTenantITHelper = getAADWebApiITHelper(AAD_MULTI_TENANT_CLIENT_ID, AAD_MULTI_TENANT_CLIENT_SECRET);
+    }
+
+    private AADWebApiITHelper getAADWebApiITHelper(String clientId, String clientSecret) {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("azure.activedirectory.client-id", clientId);
+        properties.put("azure.activedirectory.client-secret", clientSecret);
+        properties.put("azure.activedirectory.user-group.allowed-groups", "group1,group2");
+        return new AADWebApiITHelper(
+            DumbApp.class,
+            properties,
+            clientId,
+            clientSecret,
+            Collections.singletonList(toFullNameScope(clientId, SCOPE_GRAPH_READ)));
+    }
+
+    @Test
+    public void testAllowedEndpointsForSingleTenantByAccessToken() {
+        assertEquals("home", singleTenantITHelper.httpGetStringByAccessToken("home"));
+        assertEquals("api/all", singleTenantITHelper.httpGetStringByAccessToken("api/all"));
+        assertEquals("api/group1", singleTenantITHelper.httpGetStringByAccessToken("api/group1"));
+    }
+
+    @Test
+    public void testAllowedEndpointsForSingleTenantByCookie() {
+        assertEquals("home",
+            singleTenantITHelper.httpGetCookieByAccessTokenThenGetStringByCookie("home", "home"));
+        assertEquals("api/all",
+            singleTenantITHelper.httpGetCookieByAccessTokenThenGetStringByCookie("home", "api/all"));
+        assertEquals("api/group1",
+            singleTenantITHelper.httpGetCookieByAccessTokenThenGetStringByCookie("home", "api/group1"));
+    }
+
+    @Test(expected = HttpClientErrorException.class)
+    public void testNotAllowedEndpointsForSingleTenant() {
+        singleTenantITHelper.httpGetStringByAccessToken("api/group2");
+    }
+
+    @Test
+    public void testAllowedEndpointsForMultiTenantByAccessToken() {
+        assertEquals("home", multiTenantITHelper.httpGetStringByAccessToken("home"));
+        assertEquals("api/all", multiTenantITHelper.httpGetStringByAccessToken("api/all"));
+        assertEquals("api/group1", multiTenantITHelper.httpGetStringByAccessToken("api/group1"));
+    }
+
+    @Test
+    public void testAllowedEndpointsForMultipleTenantByCookie() {
+        assertEquals("home",
+            multiTenantITHelper.httpGetCookieByAccessTokenThenGetStringByCookie("home", "home"));
+        assertEquals("api/all",
+            multiTenantITHelper.httpGetCookieByAccessTokenThenGetStringByCookie("home", "api/all"));
+        assertEquals("api/group1",
+            multiTenantITHelper.httpGetCookieByAccessTokenThenGetStringByCookie("home", "api/group1"));
+    }
+
+    @Test(expected = HttpClientErrorException.class)
+    public void testNotAllowedEndpointsForMultiTenant() {
+        multiTenantITHelper.httpGetStringByAccessToken("api/group2");
```

Comment:
`api/notExist` here might be better?

Reply:
Thank you for your check. I updated it just now.